stripe/src/main/java/com/stripe/android/model/Customer.java

```diff
@@ -0,0 +1,117 @@
+package com.stripe.android.model;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.stripe.android.util.StripeNetworkUtils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.stripe.android.util.StripeJsonUtils.optString;
+import static com.stripe.android.util.StripeJsonUtils.putStringIfNotNull;
+
+/**
+ * Model for a Stripe Customer object
+ */
+public class Customer extends StripeJsonModel {
+
+    private static final String FIELD_ID = "id";
+    private static final String FIELD_OBJECT = "object";
+    private static final String FIELD_DEFAULT_SOURCE = "default_source";
+    private static final String FIELD_SHIPPING = "shipping";
+    private static final String FIELD_SOURCES = "sources";
+
+    private static final String VALUE_OBJECT = "customer";
+
+    private String mId;
+    private String mDefaultSource;
+    private ShippingInformation mShippingInformation;
+    private CustomerSources mCustomerSources;
```

Comment:
> The reason for not making it a list from the start was that it was part of a sub-object with the has_more / total_count / url fields. I could make it an inner class that's invisible to the user (so they could call Customer#getSourcesTotalCount or something of the sort).

Yep, `sources` are returned as a list object [0]. I don't think we necessarily need to follow this structure in the SDK, though, because most customers won't have that many sources. I think it might be better to just ignore the list object, and just represent `sources` as an array of `StripePaymentSource`s. 

[0] https://stripe.com/docs/api#pagination